Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/NodeTest.java

```diff
@@ -0,0 +1,1134 @@
+        }
+    }
+
+    /**
+     * Tests {@code containsKey(IdentifiableNode)}.
+     */
+    @Test
+    public void testContainsKeyIdentifiableNode() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Literal javac = storage.createLiteral("javac.exe", RDF.PLAIN_LITERAL);
+            Node mods = storage.createNode(Mods.MODS);
+            NodeReference kitodo = storage.createNodeReference("https://www.kitodo.org/");
+
+            Node node = storage.createNode().add(javac).add(kitodo).put(Mets.TYPE, mods);
+
+            assertThat(node.containsKey(Mets.TYPE), is(equalTo(true)));
+        }
+    }
+
+    /**
+     * Tests {@code containsKey(String)}.
+     */
+    @Test
+    public void testContainsKeyString() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Literal javac = storage.createLiteral("javac.exe", RDF.PLAIN_LITERAL);
+            Node mods = storage.createNode(Mods.MODS);
+            NodeReference kitodo = storage.createNodeReference("https://www.kitodo.org/");
+
+            Node node = storage.createNode().add(javac).add(kitodo).put(Mets.TYPE, mods);
+
+            assertThat(node.containsKey(Mets.TYPE.getIdentifier()), is(equalTo(true)));
+        }
+    }
+
+    /**
+     * Tests {@code contains(Object)}.
+     */
+    @Test
+    public void testContainsObject() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Literal javac = storage.createLiteral("javac.exe", RDF.PLAIN_LITERAL);
+            Node mods = storage.createNode(Mods.MODS);
+            NodeReference kitodo = storage.createNodeReference("https://www.kitodo.org/");
+
+            Node node = storage.createNode().add(javac).add(kitodo).put(Mets.TYPE, mods);
+
+            assertThat(node.contains(mods), is(equalTo(true)));
+            assertThat(node.contains(javac), is(equalTo(true)));
+            assertThat(node.contains(kitodo), is(equalTo(true)));
+        }
+    }
+
+    /**
+     * Tests {@code equals(Object)} for two different nodes.
+     */
+    @Test
+    public void testEqualsForTwoDifferentNodes() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Node one = storage.createNode(Mets.METS);
+            Node other = storage.createNode(Mods.DISPLAY_FORM);
+
+            assertThat(one, is(not(equalTo(other))));
+        }
+    }
+
+    /**
+     * Tests {@code equals(Object)} for two equal nodes.
+     */
+    @Test
+    public void testEqualsForTwoEqualNodes() {
+        for (Storage storage : TestConfig.STORAGES_TO_TEST_AGAINST) {
+            Node one = storage.createNode(Mets.METS);
+            Node another = storage.createNode(Mets.METS);
+
+            assertThat(one, is(equalTo(another)));
+        }
+    }
+
+    /**
+     * Tests {@code find(Node)}.
+     */
+    @Test
+    public void testFind() {
+        new GraphPathTest().testApplyingAGraphPathToANode();
```

Comment:
What did you test here? It did not look like a test.

Reply:
The test in class NodeTest invokes the Test in class GraphPathTest which makes use of the find method that I want to test.

Reply:
Is it not violation of rule that tests should be independent of each other?